### PR TITLE
Replaced internal `SourceIndex` with Ralphc's implementation

### PR DESCRIPTION
**Pull request by simerplaha**

### CODE REVIEW

File: compiler-access/src/main/scala/org/alephium/ralph/lsp/access/compiler/message/error/StringError.scala [hunk 1]
```diff
@@ -1,12 +1,13 @@
...
-import org.alephium.ralph.lsp.access.compiler.message.{CompilerMessage, SourceIndex}
+import org.alephium.ralph.SourceIndex
+import org.alephium.ralph.lsp.access.compiler.message.{CompilerMessage, SourceIndexExtra}
 
 object StringError {
   @inline def apply(message: String): StringError =
     StringError(
       message = message,
-      index = SourceIndex.empty
+      index = SourceIndexExtra.zero
```
Comment (tdroxler) [inline compiler-access/src/main/scala/org/alephium/ralph/lsp/access/compiler/message/error/StringError.scala:10]: why not just keeping `SourceIndex.empty`?

Comment (simerplaha) [reply compiler-access/src/main/scala/org/alephium/ralph/lsp/access/compiler/message/error/StringError.scala:10]: `SourceIndex.empty` creates a new instance on every call. `SourceIndexExtra.zero` is final.

Comment (tdroxler) [reply compiler-access/src/main/scala/org/alephium/ralph/lsp/access/compiler/message/error/StringError.scala:10]: ooooh indeed, I guess  I'll make it final in the node too.

**Review by tdroxler — APPROVED** [review]

File: compiler-access/src/main/scala/org/alephium/ralph/lsp/access/compiler/message/error/StringError.scala (hunk 1 above)
Comment (simerplaha) [reply compiler-access/src/main/scala/org/alephium/ralph/lsp/access/compiler/message/error/StringError.scala:10]: That'll be great. Thank you.

**Pull request merged**